src/Clap-Core/ClapContext.class.st

```diff
@@ -16,28 +16,40 @@ Class {
 		'stdio',
 		'match',
 		'leftovers',
-		'session'
+		'session',
+		'exitException'
 	],
 	#category : #'Clap-Core-Activation'
 }
 
 { #category : #accessing }
 ClapContext class >> defaultRoot [
-	^ (ClapRoot withName: 'clap')
+	^ (ClapRoot id: #clap)
 		description: 'Entry point for commands implemented with Clap';
 		add: ClapFlag forHelp;
 		add: ClapCommand forHelp;
+
+		add: ((ClapFlag id: #save)
+			description: 'Save the image after the command terminates');
+		add: ((ClapFlag id: #noSave)
+			description: 'Do not save the image even if the command is stateful');
+		add: ((ClapFlag id: #rename)
+			description: 'Rename the image as <NAME> before running the command';
+			add: (ClapPositional id: #NAME));
+
+		add: ((ClapFlag id: #keepAlive)
+			description: 'Keep the image running after the command terminates');
+
 		meaning: [ :args |
-			args atName: 'help' ifFound: [ :help |
+			args at: #helpFlag ifPresent: [ :help |
 				help value; exitSuccess ].
 			args validateAll ];
 		yourself
-	"setup default flags, omnipresent subcommands & basic meanings here"
 ]
 
 { #category : #accessing }
 ClapContext class >> pragmaCommands [
-	^ (PragmaCollector filter: [:prg | prg selector = 'commandline']) reset
+	^ (PragmaCollector filter: [:prg | prg selector = #commandline]) reset
```

Comment:
Be careful, the pragma collector is super slow in large images.
I think this would be ok for now, but we should avoid using it in the mid-term.

Reply:
How do we use pragmas then?

Reply:
I don't have a proper answer for that yet.
The pragma collector does a full scan of the system, which is not good in large systems.
Moreover, several usages on the system were re-scanning everything on each method compilation.
As I said, I'm ok with this for now, but we should think an alternative in the mid term.

For example, we should be able to scope the pragma collector on a hierarchy.
That will be already something.
Another thing would be to use a pragma collector that updates itself with announcements, so paying the full-scan price only once.

Maybe @tesonep has more input.

Reply:
I think we agree that pragmas are here to stay, and even if they need work, it's ouf of the scope of this PR?

Reply:
That's what I said. What is there to argue?
Let me re-quote myself:
> I think this would be ok for now, but we should avoid using it in the mid-term.

Reply:
That can be read as *Clap might have to pivot to another completely different mechanism in the future* ; of course `PragmaCollector` might go away, but I'm pretty sure there will still be some convenient API to find all methods with a given pragma, so I expect the change to be relatively simple as far as Clap is concerned.

If required, Clap specifications could be built once and installed in some static global (which is one of the reasons I don't like the meaning blocks closing over the builder, in the pillar command)

Reply:
> That can be read as _Clap might have to pivot to another completely different mechanism in the future_

And that's what I meant. Where "in the future" = in the following months, very probably during Pharo 9.

What I meant is "be careful, this may change soon, but for this PR is ok".
Is that ok?